tests/mcpdotnet.Tests/ClientIntegrationTests.cs

```diff
@@ -382,4 +405,69 @@ public async Task CallTool_Stdio_MemoryServer()
 
         await client.DisposeAsync();
     }
+
+    [Fact]
+    public async Task GetAIFunctionsAsync_UsingEverythingServer_ToolsAreProperlyCalled()
+    {
+        if (s_openAIKey is null)
+        {
+            return; // Skip the test if the OpenAI key is not provided
+        }
+
+        // Get the MCP client and tools from it.
+        await using var client = await McpClientFactory.CreateAsync(_fixture.EverythingServerConfig, _fixture.DefaultOptions); ;
+        var mappedTools = await client.GetAIFunctionsAsync();
+
+        // Create the chat client.
+        using IChatClient chatClient = new OpenAIClient(s_openAIKey).AsChatClient("gpt-4o-mini")
+            .AsBuilder()
+            .UseFunctionInvocation()
+            .Build();
+
+        // Create the messages.
+        List<ChatMessage> messages = [new(ChatRole.System, "You are a helpful assistant.")];
+        if (client.ServerInstructions is not null)
+        {
+            messages.Add(new(ChatRole.System, client.ServerInstructions));
+        }
+        messages.Add(new(ChatRole.User, "Please call the echo tool with the string 'Hello MCP!' and output the response ad verbatim."));
+
+        // Call the chat client
+        var response = await chatClient.GetResponseAsync(messages, new() { Tools = [.. mappedTools], Temperature = 0 });
+
+        // Assert
+        Assert.Equal("Echo: Hello MCP!", response.Text);
+    }
+
+    [Fact]
+    public async Task SamplingViaChatClient_RequestResponseProperlyPropagated()
+    {
+        if (s_openAIKey is null)
+        {
+            return; // Skip the test if the OpenAI key is not provided
```

Comment:
We might want to consider using xunit.v3 which now has native support for skipping unit tests. The upgrade path is fairly easy, I've already done it in my personal projects.